src/QMCWaveFunctions/LCAO/SoaAtomicBasisSet.h

```diff
@@ -780,40 +780,42 @@ class SoaAtomicBasisSet
     auto& correctphase = atom_bs_leader.mw_mem_handle_.getResource().correctphase;
     correctphase.resize(nElec);
 
-    auto* dr_ptr = dr.data();
-    auto* r_ptr  = r.data();
+    auto* correctphase_ptr                 = correctphase.device_data();
+    auto* periodic_image_displacements_ptr = periodic_image_displacements_.device_data();
+    auto* dr_ptr                           = dr.device_data();
+    auto* r_ptr                            = r.device_data();
 
-    auto* correctphase_ptr = correctphase.data();
+    ///Updated From Distance table in SoaLocalizedBasisSet.cpp
+    auto* displ_list_ptr = displ_list.device_data();
+    auto* Tv_list_ptr    = Tv_list.device_data();
 
-    auto* Tv_list_ptr    = Tv_list.data();
-    auto* displ_list_ptr = displ_list.data();
 
     constexpr RealType cone(1);
     constexpr RealType ctwo(2);
 
+
+    ///Assumed to be already on DEVICE
     //V,Gx,Gy,Gz,L
-    auto* restrict psi_ptr    = psi_vgl.data_at(0, 0, 0);
-    auto* restrict dpsi_x_ptr = psi_vgl.data_at(1, 0, 0);
-    auto* restrict dpsi_y_ptr = psi_vgl.data_at(2, 0, 0);
-    auto* restrict dpsi_z_ptr = psi_vgl.data_at(3, 0, 0);
-    auto* restrict d2psi_ptr  = psi_vgl.data_at(4, 0, 0);
+    auto* restrict psi_ptr    = psi_vgl.device_data_at(0, 0, 0);
+    auto* restrict dpsi_x_ptr = psi_vgl.device_data_at(1, 0, 0);
+    auto* restrict dpsi_y_ptr = psi_vgl.device_data_at(2, 0, 0);
+    auto* restrict dpsi_z_ptr = psi_vgl.device_data_at(3, 0, 0);
+    auto* restrict d2psi_ptr  = psi_vgl.device_data_at(4, 0, 0);
 
     {
       ScopedTimer local_timer(phase_timer_);
-#if not defined(QMC_COMPLEX)
+#if !defined(QMC_COMPLEX)
 
-      PRAGMA_OFFLOAD("omp target teams distribute parallel for map(to:correctphase_ptr[:nElec]) ")
+      PRAGMA_OFFLOAD("omp target teams distribute parallel for \
+                      is_device_ptr(correctphase_ptr) ")
       for (size_t i_e = 0; i_e < nElec; i_e++)
-        correctphase_ptr[i_e] = 1.0;
-
+        correctphase_ptr[i_e] = RealType(1.0);
 #else
-      auto* SuperTwist_ptr = SuperTwist.data();
 
-      PRAGMA_OFFLOAD("omp target teams distribute parallel for map(to:SuperTwist_ptr[:SuperTwist.size()], \
-		      Tv_list_ptr[3*nElec*center_idx:3*nElec], correctphase_ptr[:nElec]) ")
+      PRAGMA_OFFLOAD(" omp target teams distribute parallel for \
```

Comment:
remove space `" omp`

Reply:
The space is still here.

Reply:
Sorry.. was removed in the other PR but not this one